shared/Microsoft.Extensions.Buffers.MemoryPool.Sources/MemoryPoolBlock.Debug.cs

```diff
@@ -24,100 +34,118 @@ internal sealed class MemoryPoolBlock : MemoryManager<byte>
         /// </summary>
         internal MemoryPoolBlock(SlabMemoryPool pool, MemoryPoolSlab slab, int offset, int length)
         {
+            _pool = pool;
+            _slab = slab;
+
             _offset = offset;
             _length = length;
-
-            Pool = pool;
-            Slab = slab;
         }
 
-        /// <summary>
-        /// Back-reference to the memory pool which this block was allocated from. It may only be returned to this pool.
-        /// </summary>
-        public SlabMemoryPool Pool { get; }
-
-        /// <summary>
-        /// Back-reference to the slab from which this block was taken, or null if it is one-time-use memory.
-        /// </summary>
-        public MemoryPoolSlab Slab { get; }
-
         public override Memory<byte> Memory
         {
             get
             {
-                if (!Slab.IsActive) MemoryPoolThrowHelper.ThrowObjectDisposedException(MemoryPoolThrowHelper.ExceptionArgument.MemoryPoolBlock);
+                if (!_slab.IsActive)
+                {
+                    MemoryPoolThrowHelper.ThrowObjectDisposedException(MemoryPoolThrowHelper.ExceptionArgument.MemoryPoolBlock);
```

Comment:
Would it be better to have an _isDisposed field in MemoryPoolBlock itself that was set in Dispose and unset in Lease? Then there'd be a different exception for using the block after it's returned to the pool and using the block before it's returned to the pool but after the pool is disposed.

Reply:
Thought about it too, will do.